catalogue/webapp/pages/search/index.tsx

```diff
@@ -342,12 +342,13 @@ export const getServerSideProps: GetServerSideProps<
         ...defaultProps,
         ...(stories && stories.pageResults?.length && { stories }),
         ...(images?.pageResults.length && { images }),
-        works: works
-          ? {
-            ...works,
-            pageResults: works.pageResults.map(toWorkBasic),
-          }
-          : {},
+        works:
+          works && works.pageResults.length
+            ? {
+                ...works,
+                pageResults: works.pageResults.map(toWorkBasic),
+              }
+            : undefined,
```

Comment:
I think we could just copy the format from the other ones above to clarify it's doing the same thing?

```
...(works && works.pageResults.length && { 
          ...works,
          pageResults: works.pageResults.map(toWorkBasic), 
}),
```

Reply:
I had a few issues with the type checker and the old behaviour – in particular it couldn't see that the types had changed in a breaking way, which is why I wrote it in a less complex way.

I should probably convert the other expressions to match (and get them properly type checked), but I'm going to leave that for a separate PR.